src/worker/core/season/awards.ts

```diff
@@ -321,11 +327,29 @@ const saveAwardsByPlayer = async (
 		}
 	}
 };
+const deleteAwardsByPlayer = async (
+	awardsByPlayer: AwardsByPlayer,
+	conditions: Conditions,
+	season: number,
+	awardsToDelete: string[],
+) => {
+	const pids = Array.from(new Set(awardsByPlayer.map(award => award.pid)));
+	for (const pid of pids) {
+		const p = await idb.cache.players.get(pid);
+		if (p) {
+			p.awards = p.awards.filter(
+				x => x.season != season || x.type! in awardsToDelete,
```

Comment:
`x.type! in awardsToDelete` - how does this even work? If `awardsToDelete` is an array, this should always return false, because `in` looks at the keys of an object (which for an array are the numeric indices) and `x.type` is not a number.

Reply:
This is so it doesn´t delete League champs. Basically what it does is just leaves those awards that are not changed on the form and those awards from other season

Reply:
I understand that's what you intend for it to do, but `x.type! in awardsToDelete` will always be false. Try and see. Probably should be `!awardsToDelete.includes(x.type)`

Reply:
Doesn´t work since awardsToDelete is an object so includes is not a method. I have seen that The `x.type! in awardsToDelete` is not always false. For example if the player has a finalChamps award or an all star award, those awards is not included in awardsToDelete so the filter applies. I have tried erasing that part of the filter  and it deletes the finals champ awards, which is what I´m trying to avoid. and I have printed in the console what `x.type!in awardsToDelete`  returns and it returns all the all star and champ awards, so that when updating the awards those are not changed whatsoever.

Reply:
`awardsToDelete: string[],` a few lines up says it's an array, and the only time this function is called, it's given a value of `Object.values(awardNames)` which is an array.